File: app/src/main/java/com/example/pam_app/db/BucketWithEntriesEntity.java
package com.example.pam_app.db;

import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;

public class BucketWithEntriesEntity {
    @Embedded
    public BucketEntity bucket;
    @Relation(parentColumn = "id", entityColumn = "idBucket")
    public List<BucketEntryEntity> entries;
}

File: app/src/main/java/com/example/pam_app/db/BucketWithEntriesEntity.kt
package com.example.pam_app.db

import androidx.room.Embedded
import androidx.room.Relation
import kotlin.jvm.Synchronized

class BucketWithEntriesEntity {
    @kotlin.jvm.JvmField
    @Embedded
    var bucket: BucketEntity? = null

    @kotlin.jvm.JvmField
    @Relation(parentColumn = "id", entityColumn = "idBucket")
    var entries: List<BucketEntryEntity>? = null
}